src/types/inc/IInputEvent.hpp

```diff
@@ -539,8 +541,15 @@ class FocusEvent : public IInputEvent
 
     void SetFocus(const bool focus) noexcept;
 
+    // BODGY - see FocusEvent.cpp for details.
+    constexpr bool CameFromApi() const noexcept
+    {
+        return _cameFromApi;
+    }
+
 private:
     bool _focus;
+    const bool _cameFromApi;

```

Comment:
Generally, don't use `const`. It prevents `std::move` and other kinds of language/compiler features, because C++ is weird.

Reply:
> It prevents `std::move` and other kinds of language/compiler features

TIL, thanks